configure.ac

```diff
@@ -200,7 +200,7 @@ AC_PATH_PROG(PERL, perl)
 #   -Wall                        CC, CXX
 #
 
-PROSPECTIVE_CFLAGS="-Wall"
+PROSPECTIVE_CFLAGS="-Wall -Werror"
```

Comment:
Great, I thought `-Wall` would enable all possible warnings. Adding these two found more warnings. It will take some time to get them fixed.

And It seems we don't need to add same flags to `PROSPECTIVE_CXXFLAGS`. It's combined with `PROSPECTIVE_CFLAGS ` at last.

> PROSPECTIVE_CFLAGS="-Werror"
PROSPECTIVE_CXXFLAGS=""
AX_CHECK_COMPILER_OPTIONS([C],   ${PROSPECTIVE_CFLAGS})
AX_CHECK_COMPILER_OPTIONS([C++], ${PROSPECTIVE_CFLAGS} ${PROSPECTIVE_CXXFLAGS})